### PR TITLE
virtio-pci block support

### PR DESCRIPTION
**Pull request by rbradford**
@sameo @sboeuf Just a preview of the work i've done today. The substantial work that remains is creating a VirtioPCITransport. I'm hoping it should be straightforward now that I have retrieved the location of the appropriate structures in the BAR from the PCI capabilities.

### CODE REVIEW

File: src/pci.rs [hunk 1]
```diff
@@ -107,6 +107,30 @@ struct PciBar {
     address: u64,
 }
 
+#[cfg(not(test))]
+enum PciCapabilityType {
```
Comment (sboeuf) [inline src/pci.rs]: Those types are really virtio based. That's why I would call this enum `VirtioPciCapabilityType`.

Following the same observation, I think we should put those into their own separate file `virtio-pci.rs`. A clear separation between PCI and virtio-pci seems important to me.

File: src/pci.rs [hunk 2]
```diff
@@ -116,6 +140,7 @@ struct PciDevice {
     bars: [PciBar; 6],
     vendor_id: u16,
     device_id: u16,
+    caps: [VirtioPciCap; PciCapabilityType::MaxCaps as usize - 1], // caps start at 1
```
Comment (sboeuf) [inline src/pci.rs]: If you're not planning to support real PCI devices, this shortcut is fine. But if we care about having a clear separation between the PCI and virtio-pci bits, we should not have this field as part of the `PciDevice`.

Comment (rbradford) [reply src/pci.rs]: Yeh my plan is to split that into the new VirtioPCITransport which implements the VirtioTransport needed for the virtio implementation.

File: src/pci.rs (hunk 1 above)
Comment (rbradford) [reply src/pci.rs]: Agreed to the latter. But i'm not going to split the file at this point as PCI is only used for virtio-pci.

**Comment by rbradford** [comment]
> Also, I guess the next step here is to filter out the devices that you found through the PCI enumeration, and pick only the ones matching virtio-blk deviceID/vendorID, right?

Exactly. This is just some proof of concept debugging stuff to check that the PCI stuff is working.

**Pull request merged**